src/modules/quick_api_module.f90

```diff
@@ -516,11 +519,11 @@ subroutine run_quick(self,ierr)
   SAFE_CALL(set_quick_molspecs(quick_api,ierr))
 
   ! start the timer for initial guess
-  call cpu_time(timer_begin%TIniGuess)
+  RECORD_TIME(timer_begin%TIniGuess)
 
   ! we will reuse density matrix for steps above 1. For the 1st step, we should
   ! read basis file and run SAD guess.
-  if(self%firstStep .and. self%reuse_dmx) then
+  if((self%firstStep .and. self%reuse_dmx) .or. .not. self%reuse_dmx) then
```

Comment:
Simpler and identical result
```
if(self%firstStep .or. (.not. self%reuse_dmx)) then
```
because we always want to do a guess if we are on the first step or if we do not want to reuse the density.

Reply:
Thanks for pointing this out!